### ISSUE TITLE
aws.1.0.2: upper bound on cohttp

### ISSUE THREAD

**Issue by jrochel** [ocaml/opam-repository#18212]
In the package `aws` there is an upper version limit (implemented as a conflict) on the cohttp package:
```
conflicts: [
  "cohttp" {< "0.17.0"}
  "cohttp" {>="0.99"}
]
```
Is there any good reason for this? Everything compiles just fine for me with cohttp.2.5.4. See also: https://github.com/inhabitedtype/ocaml-aws/issues/96

**Comment by kit-ty-kate** [comment]
The constraint was added 4 years ago in https://github.com/ocaml/opam-repository/pull/9685, probably because the API had changed significantly. I'm not sure if it was tested as failing with it.

**Comment by kit-ty-kate** [comment]
Ok, i had a look. The package itself indeed does compile. However it is for the wrong reason.
The archive contains calls to `cohttp.lwt` which is not available with cohttp >= 1.0: https://github.com/inhabitedtype/ocaml-aws/blob/d6a1c64e5ba1e63aa6d84157e427e4e9bce27ff3/_tags#L27
What is happening is that the depopt is not completely correct and requires `cohttp` + `lwt` + `ssl` to be installed for this code to be compiled: https://github.com/ocaml/opam-repository/blob/master/packages/aws/aws.1.0.2/opam#L25

So by simply adding `ssl` to the mix, I get:
```
[opam@500043e6b648 ~]$ opam install cohttp.2.5.4 aws.1.0.2 lwt ssl

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[aws.1.0.2] synchronised (https://github.com/inhabitedtype/ocaml-aws/releases/download/aws-1.0.2/aws-1.0.2.tar.gz)

[NOTE] Package lwt is already installed (current version is 5.4.0).
[NOTE] Package aws is already installed (current version is 1.0.2).
[NOTE] Package cohttp is already installed (current version is 2.5.4).
The following actions will be performed:
  ∗ install   conf-libssl 2      [required by ssl]
  ∗ install   ssl         0.5.9
  ↻ recompile aws         1.0.2*
===== ∗ 2   ↻ 1 =====

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
∗ installed conf-libssl.2
⬇ retrieved aws.1.0.2  (https://github.com/inhabitedtype/ocaml-aws/releases/download/aws-1.0.2/aws-1.0.2.tar.gz)
⬇ retrieved ssl.0.5.9  (https://github.com/savonet/ocaml-ssl/archive/0.5.9.tar.gz)
⊘ removed   aws.1.0.2
∗ installed ssl.0.5.9
[ERROR] The compilation of aws.1.0.2 failed at "ocaml setup.ml -configure --prefix /home/opam/.opam/4.11 --enable-lwt --disable-async".

#=== ERROR while compiling aws.1.0.2 ==========================================#
# context     2.1.0~beta4 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/inhabitedtype/ocaml-aws/releases/download/aws-1.0.2/aws-1.0.2.tar.gz)
# path        ~/.opam/4.11/.opam-switch/build/aws.1.0.2
# command     ~/.opam/4.11/bin/ocaml setup.ml -configure --prefix /home/opam/.opam/4.11 --enable-lwt --disable-async
# exit-code   1
# env-file    ~/.opam/log/aws-8236-27bf8b.env
# output-file ~/.opam/log/aws-8236-27bf8b.out
### output ###
[...]
# ocamlfind: Package `cohttp.lwt' not found
# W: Field 'pkg_cohttp_lwt' is not set: Command ''/home/opam/.opam/4.11/bin/ocamlfind' query -format %d cohttp.lwt > '/tmp/oasis-4d1cab.txt'' terminated with error code 2
# E: Cannot find findlib package cohttp.lwt
# E: Failure("1 configuration error")
```

**Comment by kit-ty-kate** [comment]
Since the mystery has been resolved I'll close this issue.
However, the latest version is 4 years old and master seems to be all good and up-to-date. I would encourage the authors to make a new release.
